File: oop04.rb
class C
    def initialize(v)
        @value = v
    end
    def show()
        p @value
    end
end

c1 = C.new(10)
# p c1.value() # value라는 메소드를 찾기 때문에 error 남 -> 즉, 루비에서 인스턴스 변수에 접근이 불가능 함
# c1.value = 20 # 메소드 밖에서 직접적으로 인스턴스 변수에 접근이 불가능함
c1.show() # 인스턴스 변수에 접근하려면 메서드 안에서 지정하여 밖에서는 메서드를 통해서 접근 가능

File: Oop/oop04.py
class C:
    def __init__(self, v):
        self.value = v

    def show(self):
        print(self.value)


c1 = C(10)
print(c1.value)  # c1에 소속된 value라고 하는 인스턴스 변수의 값을 읽어와 -> 인스턴스 변수에 접근을 허용함
c1.value = 20  # 읽어 왔으면 쓰기도 가능함
print(c1.value)
c1.show()  # 메서드 안에서 지정하여 밖에서는 메서드를 통해서 접근하는것도 가능
